README.md

```diff
@@ -256,28 +256,34 @@ Note: This requirement and the requirement to only use secure protocols makes pa
+After unpartitioned third-party cookies are removed, a third party will need to set one cookie per top-level context that the user visits, resulting in more cookies set on user's machines.
 
 Browsers that wish to support partitioned cookies must impose additional limitations on the number of cookies available to a third-party domain per-partition.
 
 However, it is also necessary for user agents to design these limits in a way that does not allow malicious third parties from learning cross-site information about users.
-See [Limit the number of cookies in a top-level context's partition](#limit-the-number-of-cookies-in-a-top-level-contexts-partition) for [Applying the 180 cookies-per-domain limit](#applying-the-180-cookies-per-domain-limit) in [Alternative Solutions](#alternative-solutions) for more details.
+See [Limit the number of cookies in a partition](#limit-the-number-of-cookies-in-a-partition) for [Applying the 180 cookies-per-domain limit](#applying-the-180-cookies-per-domain-limit) in [Alternative Solutions](#alternative-solutions) for more details.
 
 ## Detailed Design
 
-### Opt-in cookie attribute
+### Partitioning model
 
-We propose a new cookie attribute, `Partitioned`, which must be specified by the `Set-Cookie` header to indicate that the cookie should be stored in a per-top-level partition.
-Any cookies that are not set with the `Partitioned` attribute will eventually be blocked in third-party contexts.
-(Note: Other features like the [`SameParty`](https://github.com/cfredric/sameparty) attribute may adjust the details of such blocking across domains within the same First-Party Set.)
+Today, cookies are keyed on the hostname or domain of the site that set them, i.e. their _host key_.
+After CHIPS, cookies that opt into partitioning will be double-keyed on their _partition key_ and their _host key_.
+A cookie's partition key is the [site](https://html.spec.whatwg.org/#sites) (i.e. scheme and registrable domain) of the top-level URL the browser was visiting at the start of the request to the endpoint that set the cookie.
 
-### Partition by top-level context
+Likewise, a request's partition key is the site of the top-level URL the browser is visiting at the start of a request.
+Browsers must only send a cookie with the `Partitioned` attribute in requests with the same partition key as that cookie.
 
-Cookies set with the `Partitioned` attribute must only be sent to the third-party host when the user agent is in the same _top-level context_ as when the cookie was first set.
-
-For most sites, the top-level context is just the top-level site. However, when the top-level site is part of a [First-Party Set](https://github.com/privacycg/first-party-sets), the third-party can share the same partition across sites within the same set.
+If the top-level site is part of a [First-Party Set](https://github.com/privacycg/first-party-sets), the third-party can share the same partition across sites within the same set.
```

Comment:
If the top-level document in an http (non-secure) URL, we don't actually allow application of FPS, because FPS is only supported in secure contexts (see https://github.com/privacycg/first-party-sets#declaring-a-first-party-set "An origin is in the first-party set if: Its scheme is https"). Can we change this to:

"If the top-level site has an https scheme and is part of a First-Party Set..."

and add:

"If the top-level site has a http scheme, the First-Party Set feature does not apply, since it is only support for origins with an https scheme."

Reply:
Done.